File: src/main/java/com/tekcay/bbouapp/aliments/SimpleAliment.java
package com.tekcay.bbouapp.aliments;

import com.tekcay.bbouapp.Utils;
import javafx.scene.control.TextField;

public record SimpleAliment(int id, String localizedName, int glucides, int proteines, int lipides, int kcal) {

    public static SimpleAliment build(int id, TextField nameField, TextField glucideField, TextField proteineField, TextField lipideField, TextField kcalField) {

        int glucides = Utils.getNumericValue(glucideField);
        int proteines = Utils.getNumericValue(proteineField);
        int lipides = Utils.getNumericValue(lipideField);
        int kcal = Utils.getNumericValue(kcalField);

        if (kcal == 0) kcal = setKcal(glucides, proteines, lipides);

        return new SimpleAliment(id, nameField.getText(), glucides, proteines, lipides, kcal);
    }

    public static int setKcal(int glucides, int proteines, int lipides) {
        return (glucides + proteines) * 4 + lipides * 9;
    }
}

File: src/main/kotlin/com/tekcay/bbouapp/SimpleAliment.kt
package com.tekcay.bbouapp;

import com.google.gson.Gson
import java.io.Writer

data class SimpleAliment(val id: Int, val name: String, val glucides: Int, val proteines: Int, val lipides: Int, val kcal: Int) {

    fun serialize(): String {
        var gson = Gson()
        return gson.toJson(this)
    }


}
